### PR TITLE
[nucleus-3.0] alert component

### PR DESCRIPTION
**Pull request by MekalaNagarajan-Centrica**
- [x] Deprecate highlighter component
- [x] Add alert component

### CODE REVIEW

File: docs/components/ns-alert.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+## Accessibility 
+
+The information contained in the alert should always be the most important information on the page. To ensure that screen reader users are informed about the importance of these messages the content is read aloud as soon as the alert type is used. This could be when the page loads or when it is dynamically added to a page.
+
+If there are multiple alerts on a page they are read:
+
+* from top to bottom on page load
+* in the order they have been added to the page when dynamically added
+
+We recommend you use ns-alert for their role, not for their looks.
```
Comment (PtitBen) [inline docs/components/ns-alert.md:70]:
```suggestion
We recommend you use `ns-alert` for their role, not for their looks.
```

File: docs/components/ns-alert.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+The `highlighter` type of `ns-alert` can be used to communicate information or error, warning and success messages.
+
+It communicates the message that a service is not available to them or that there are technical issues on specific pages only. These messages are different to global messages as they only apply to some pages not site wide.
+
+e.g. a message that says the top up service is not available would only apply to pages that customers will look at in order to top up.
+
+
+### Content guidance
+
+![Alert - Highlighter - Error state](images/ns-highlighter/content-guidance.webp)
```
Comment (PtitBen) [inline docs/components/ns-alert.md:26]:
```suggestion
#### Highlighter variant

![Alert - Highlighter - Error state](images/ns-highlighter/content-guidance.webp)
```

File: docs/components/ns-alert.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import { ComponentPlacement } from '../../includes/component-placement.js'
+import { PageFooter } from '../../includes/page-footer.js'
+
+## Introduction
+
+> Alert can be used to communicate contextual messages to the customers.
+
+Alert can be a highlighted message, toast or notification specific to the user or to the page the user is navigated.
+
+## Highlighter Type
```
Comment (PtitBen) [inline docs/components/ns-alert.md:15]:
```suggestion
## Highlighter variant
```

File: docs/components/ns-alert.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+The `highlighter` type of `ns-alert` can be used to communicate information or error, warning and success messages.
```
Comment (PtitBen) [inline docs/components/ns-alert.md:17]:
```suggestion
The `highlighter` variant of `ns-alert` can be used to communicate information or error, warning and success messages.
```

File: docs/components/ns-alert.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+![Alert - Highlighter - Error state](images/ns-highlighter/content-guidance.webp)
+
+| Key | Field type | Guidelines |
+| :--- | :--- | :--- |
+| A | Icon | This is the icon inside the `highlighter` type. This is dictated by the `state` of Alert. The [specification table](#specification) has a list of options. |
+| B | Heading | The recommended length is between 4 and 12 words, not exceeding 50 characters in total. |
+| C | Content | A single, short paragraph works best. You are able to use inline text links within the paragraph if necessary. Don't add additional headings within the paragraph and the copy length should not exceed more than two paragraphs of 3 lines each. Can contain bold copy `<b>`, inline links `<a>`, and a caveat at the end of the relevant paragraph if required `<a href="#caveat">1</a>`. |
+| D | Accent | This is to support the context of the messaging. This colour matches that of the icon. |
+
+The content for `highlighter` type is dependent on the state that is used:
```
Comment (PtitBen) [inline docs/components/ns-alert.md:35]:
```suggestion
The content for the  `highlighter` variant is dependent on the state that is used:
```

File: docs/components/ns-alert.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+### Considerations of best practice
+
+* Be clear and concise
+
+The message must be in plain English, and help the user out of the situation by providing an actionable link if possible.
+
+## Best practice
+
+| 💚 Do's | 💔 Don'ts |
```
Comment (PtitBen) [inline docs/components/ns-alert.md:80]:
```suggestion
### Highlighter variant

| 💚 Do's | 💔 Don'ts |
```

**Review by PtitBen — COMMENTED** [review]
I added a few suggestions to match the vocabulary we used around variants (looked at the latest component with variants we documented: ns-timeline)

File: docs/components/ns-alert.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+import { StorybookStory } from '../../includes/storybook-story.js'
+import { ComponentPlacement } from '../../includes/component-placement.js'
+import { PageFooter } from '../../includes/page-footer.js'
+
+## Introduction
+
+> Alert can be used to communicate contextual messages to the customers.
+
+Alert can be a highlighted message, toast or notification specific to the user or to the page the user is navigated.
```
[the remark this answers is not in the document]
Comment (MekalaNagarajan-Centrica) [reply docs/components/ns-alert.md:13]: This is in line with most of the other components.

File: docs/components/ns-alert.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+description: Organism | Alert component.
+---
+
+import { StorybookStory } from '../../includes/storybook-story.js'
+import { ComponentPlacement } from '../../includes/component-placement.js'
+import { PageFooter } from '../../includes/page-footer.js'
+
+## Introduction
+
+> Alert can be used to communicate contextual messages to the customers.
```
[the remark this answers is not in the document]
Comment (MekalaNagarajan-Centrica) [reply docs/components/ns-alert.md:11]: This is in line with most of the other components.
